src/sentry/static/sentry/app/views/settings/components/settingsLayout.tsx

```diff
@@ -14,18 +19,88 @@ type Props = {
   children: React.ReactNode;
 } & RouteComponentProps<{}, {}>;
 
-function SettingsLayout(props: Props) {
-  const {params, routes, route, router, renderNavigation, children} = props;
+type State = {
+  navVisible: boolean;
+  navOffsetTop: number;
+};
+
+class SettingsLayout extends React.Component<Props, State> {
+  static propTypes = {
+    renderNavigation: PropTypes.func,
+    route: PropTypes.object,
+    router: PropTypes.object,
+    routes: PropTypes.array,
+  };
+
+  constructor(props: Props) {
+    super(props);
+
+    this.headerRef = React.createRef();
```

Comment:
I would move this to the line where you have `headerRef: React.RefObject<HTMLDivElement>;` you could write it as t`his.headerRef = React.createRef<HTMLDivElement>();`